restore.py

```diff
@@ -59,6 +59,65 @@ def restoreFromBackup(backup, progress=lambda x: ()):
     except:
         pass
 
+    # should we restore old schema and move to MBR?
+    restore_partitions = None
+    if 'BOOT' not in backup_partition_layout and 'LOG' not in backup_partition_layout:
+        new_restore_partition = restore_partition
+        efi_boot = False
+
+        # remove unneeded partitions
+        if dsk.swap[0]:
+            tool.deletePartition(tool.partitionNumber(dsk.swap[1])) # swap
+        tool.deletePartitionIfPresent(boot_partnum) # boot
+
+        # use logs as root
+        root_partnum = tool.partitionNumber(dsk.root[1])
+        backup_partnum = tool.partitionNumber(backup_partition)
+        if dsk.logs[0]:
+            logs_partnum = tool.partitionNumber(dsk.logs[1])
+            if tool.partitionStart(logs_partnum) < tool.partitionStart(root_partnum) and tool.partitionSize(logs_partnum) >= (3500 << 20):
+                restore_partition = dsk.logs[1]
+                tool.deletePartition(root_partnum)
+                tool.renamePartition(logs_partnum, root_partnum)
+            else:
+                tool.deletePartition(tool.partitionNumber(dsk.logs[1])) # logs
+
+        # rewrite with MBR format, we need to manually convert IDs from GPT to MBR
```

Comment:
If there was no BOOT partition it couldn't have been GPT, or at least not if using grub

Reply:
After a discussion with Ross we decided to use MBR only if we upgraded from MBR checking a file in `/var/preserve`.
In case of GPT we need to install grub2 on the partition instead of the disk (there's an option in the answerfile, we need to update it automatically).

Reply:
Checked and implemented but looks like `grub-install` does not do a great job restoring on the partition making the system not bootable